test/unit/spec/kbaseNarrativeSpec.js

```diff
@@ -1,20 +1,163 @@
+                    comm_info: (comm_name, callback) => {
+                        callback({content: {
+                            comms: {
+                                'some_comm_id': {
+                                    target_name: 'KBaseJobs'
+                                }
+                            }
+                        }});
+                    },
+                    comm_manager: {
+                        register_comm: () => {}
+                    },
+                    execute: (code, callbacks) => {
+                        callbacks.shell.reply({
+                            content: {}
+                        });
+                    }
+                },
+                notebook_name: DEFAULT_NOTEBOOK_NAME
+            };
+            Jupyter.keyboard_manager = Jupyter.notebook.keyboard_manager;
+
+            // The NarrativeLogin.init call invokes both of the above token/user profile calls.
+            // It's called before the creation of the Narrative object. So that needs to happen here.
+            await NarrativeLogin.init(loginDiv);
+        });
+
+        afterEach(() => {
+            jasmine.Ajax.uninstall();
+            Jupyter.notebook = null;
+        });
+
+        it('Should instantiate', () => {
+            const narr = new Narrative();
+            expect(narr.maxNarrativeSize).toBe('10 MB');
+        });
+
+        it('Should have an init function that responds when the kernel is connected', () => {
+            return new Promise((resolve, reject) => {
+                const narr = new Narrative();
+                const jobsReadyCallback = (err) => {
+                    if (err) {
+                        reject('This should not have failed', err);
+                    }
+                    else {
+                        resolve();
+                    }
+                };
+                narr.init(jobsReadyCallback);
+                $([Jupyter.events]).trigger('kernel_connected.Kernel');
+            });
         });
+
+        it('init should fail as expected when the job connection fails', () => {
+            return new Promise((resolve, reject) => {
+                Jupyter.notebook.kernel.comm_info = () => {
+                    throw new Error('an error happened');
+                };
+                const narr = new Narrative();
+                const jobsReadyCallback = (err) => {
+                    if (err) {
+                        resolve();
+                    }
+                    else {
+                        reject('expected an error');
+                    }
+                };
+                narr.init(jobsReadyCallback);
+                $([Jupyter.events]).trigger('kernel_connected.Kernel');
+            });
+        });
+
+        it('Should return a boolean for is_loaded', () => {
+            const narr = new Narrative();
+            expect(narr.isLoaded()).toEqual(DEFAULT_FULLY_LOADED);  // default as set in
```

Comment:
```
// default as set in
```

unfinished comment?

Reply:
I choose to have it be dangling. Where could it be set!? I want the reader to have the same anxiety I feel when I run these tests.

(ok, I'll fix it. Thanks for catching.)